### DISCUSSION TITLE
Question: Deserialize ODataActionParameters into a C# object

### DISCUSSION THREAD

**Discussion by Tiberriver256** [OData/AspNetCoreOData#1179] [Q&A]
Is there a way to easily convert an ODataActionParameters object into an actual C# complex object?

The examples [provided here](https://learn.microsoft.com/en-us/odata/webapi-8/fundamentals/action-routing?tabs=net60%2Cvisual-studio) work great for a simple set of parameters, but what if I have a more complex object?

I know I can probably use something like [this](https://stackoverflow.com/a/64146795/3317144) since it's just a `Dictionary<string, Object>`, but I'm wondering if there's an officially supported path.

**Comment by julealgon** [comment]
I don't think there is a supported path to do this today. I've proposed the following which I believe would resolve the issue:
- https://github.com/OData/AspNetCoreOData/issues/848
